aten/src/ATen/core/jit_type.h

```diff
@@ -661,6 +661,14 @@ struct CAFFE2_API TensorType : public Type {
     return scalar_type_ && device_ && sizes_.isComplete() && strides_.isComplete();
   }
 
+  bool isInferredType() const {
+    return is_inferred_type_;
+  }
+
+  void setToInferredType(){
+    is_inferred_type_ = true;
+  }
```

Comment:
I don't think we should expose this. It opens up the possibility that a `TensorType` instance will be created with `is_inferred_type` set to `false`, and then it will be set to `true` later in some other part of the code, perhaps by mistake. Given that many `Type` instances are shared using `TypePtrs` (for example, the instance returned by `TensorType::get()`), this can be difficult to debug.

We always know whether or not a `Tensor` type was inferred when creating/retrieving a pointer to a `TensorType` instance. Accordingly, I suggest creating a function like `TensorType::getInferred()` so that we can have a singleton for all of the uses of inferred `TensorType`.